### ISSUE TITLE
Empty allow/exclude list overrides config in the API

### ISSUE THREAD

**Issue by swtwsk** [sqlfluff/sqlfluff#2560]
## Expected Behaviour
Calling API's `sqlfluff.fix` or `sqlfluff.lint` with `config_path` argument pointing to `.sqlfluff` with `exclude_rules` set should return list of errors without rules listed in `.sqlfluff`.

## Observed Behaviour
`sqlfluff.lint` and `sqlfluff.fix` both uses [get_simple_config](https://github.com/sqlfluff/sqlfluff/blob/cd8aceb5107abaf3639c4c1e2f7cca25aa193cc3/src/sqlfluff/api/simple.py#L13) to get the `FluffConfig`. Not only `get_simple_config` calls `FluffConfig.from_root` with `ignore_local_config=True` argument, but also builds up `overrides` dictionary:
```py
overrides = {
    "dialect": dialect,
    "rules": ",".join(rules) if rules is not None else None,
    "exclude_rules": ",".join(exclude_rules) if exclude_rules is not None else None,
}
```
which gets higher priority than the config referenced by a `config_path` argument. In the effect, if no `rules` or `exclude_rules` get passed to `lint` or `fix` functions, `overrides` overrides those config settings with `None` values, effectively allowing for all the rules.

## Steps to Reproduce
Add .sqlfluff file to your project:
```ini
[sqlfluff]
exclude_rules = L027,L029
```
and then call Python function:
```py
import sqlfluff
import pathlib

sql = "SELECT TRIM(name) AS name FROM some_table"
lint_results = sqlfluff.lint(sql, config_path=str(pathlib.Path.cwd().joinpath(".sqlfluff")))
for lint_result in lint_results:
  if lint_result["code"] in ["L027", "L029"]:
    raise Exception(lint_result["code"])
```

## Dialect
--

## Version
sqlfluff, version 0.9.0

However, reading the freshest code on GitHub in [simple.py](https://github.com/sqlfluff/sqlfluff/blob/cd8aceb5107abaf3639c4c1e2f7cca25aa193cc3/src/sqlfluff/api/simple.py#L13) and [config.py](https://github.com/sqlfluff/sqlfluff/blob/cd8aceb5107abaf3639c4c1e2f7cca25aa193cc3/src/sqlfluff/core/config.py#L454) it looks like the problem persists.

**Comment by barrywhart** [comment]
The code looks correct to me. If you don't want to override the config, you should pass `None` rather than `[]`. If you specify `rules` as an empty list, that's saying, "I want these 0 zero rules enabled."
